### DISCUSSION TITLE
How can I post  long text and attach image?

### DISCUSSION THREAD

**Discussion by bigturtle88** [telegraf/telegraf#1602] [General]
How can I place long text and attach an image at the bottom?

Faced such a problem. I can't find a solution. Maybe I'm using the wrong function?

 `const imageFile = fs.readFileSync('./img.png');
ctx.telegram.sendMessage(
    userId,
    'text' + imageFile,
  )`

**Comment by MKRhere** [comment]
"Attach an image at the bottom" is not a feature of Telegram. You'll need to send them as two separate messages.

```TypeScript
ctx.telegram.sendMessage(userId, text);
ctx.telegram.sendPhoto(userId, { source: fs.readFileSync('./img.png') });
```

**Reply by bigturtle88** [reply]
Thanks for the answer. How can I achieve a similar effect? For the image to be at the bottom I need to use markdown.

**Reply by MKRhere** [reply]
Telegram Markdown does not support images. Check:
https://core.telegram.org/bots/api#markdownv2-style

It's not possible to have image at the bottom of a message (except as an image preview).

**Comment by suntong** [comment]
As @MKRhere answered, the image cannot be at the bottom, but

If you're OK with the image at the top, then it is feasible to send long text and attach an image at the same time with _a single command_. 

I used to be able to do that with another even simpler telegram client. The trick is to put your long text as the caption, and the added bonus is that Markdown is supported in the caption.

(I'm still trying to find out how to do that with telegraf...)

**Reply by MKRhere** [reply]
Just pass caption in the second parameter:

```TS
ctx.sendPhoto(Input.fromLocalFile("./img.jpg"), { caption: "This is a caption" });
```

**Reply by heiheshang** [reply]
caption size is limited - this will not work if you have long text

**Reply by jjrh92** [reply]
> caption size is limited - this will not work if you have long text

Do you know what is the limit size for the caption? and if there is a workaround for that?
